errors.go

```diff
@@ -2,6 +2,29 @@ package fargo
 
 // MIT Licensed (see README.md) - Copyright (c) 2013 Hudl <@Hudl>
 
+import (
+	"fmt"
+)
+
+type unsuccessfulHTTPResponse struct {
+	statusCode    int
```

Comment:
Please take a look at commit 4f48c58, in which I make sure to treat the message prefix as optional. Given that, let me know if you'd still prefer the field order to be swapped.